int32 node
bool isVisited
